### PR TITLE
Improved code style and fixed typos

### PR DESCRIPTION
**Pull request by michaelehab**
## What
I modified the datasources files to follow PEP 8 more closely. They were already following it but they missed some small details like the docstrings. I also fixed some typos found in the vulntotal_utils.py file and other files.

## Why
I made these changes to improve the readability and consistency of the code. Following PEP 8 helps to maintain a high standard of code quality and style. Fixing typos helps to avoid confusion and errors.

## Anything else
Please let me know if you have any questions or feedback. Thank you for reviewing my PR!

### CODE REVIEW

**Comment by keshav-space** [comment]
To ensure consistent code formatting, we use Black and isort. To auto-format the code, run the `make valid` command.

Add a Developer Certificate of Origin (DCO) to your commit. This certifies that you have the right to submit the code under an open-source license.

How to write good commit messages? Refer to this guide: https://aboutcode.readthedocs.io/en/latest/contributing/writing_good_commit_messages.html

File: vulntotal/datasources/gitlab.py [hunk 1]
```diff
@@ -148,13 +168,20 @@ def get_casesensitive_slug(path, package_slug):
...
 
 
 def parse_interesting_advisories(location, version, delete_download=False) -> Iterable[VendorData]:
+    """
+    Parses advisories from YAML files in a given location that match a given version.
+    :param location: A Path object representing the location of the YAML files.
+    :param version: A string representing the version to check against the affected range.
+    :param delete_download: A boolean indicating whether to delete the downloaded files after parsing.
+    :return: An iterable of VendorData objects containing the advisory information.
+    """
```
[the remark this answers is not in the document]
Comment (michaelehab) [reply vulntotal/datasources/gitlab.py]: Changed it, and used the same format for all docstrings

**Comment by TG1999** [comment]
@keshav-space please review when you get chance

File: vulntotal/datasources/snyk.py [hunk 2]
```diff
@@ -142,7 +192,7 @@ def parse_html_advisory(advisory_html, snyk_id, affected) -> VendorData:
         if "Upgrade" in fixed:
             lower = fixed.index("version") if "version" in fixed else fixed.index("versions")
             upper = fixed.index("or")
-            fixed_versions = "".join(fixed[lower + 1 : upper]).split(",")
+            fixed_versions = [ver.strip(",") for ver in fixed[lower + 1 : upper]]
```
Comment (keshav-space) [inline vulntotal/datasources/snyk.py:195]: Revert this change, this will not reproduce the same result as the original code.
```suggestion
            fixed_versions = "".join(fixed[lower + 1 : upper]).split(",")
```

**Review by keshav-space — CHANGES_REQUESTED** [review]
Thanks, @michaelehab
See the few suggestions below, and also need to resolve the conflicts before we can merge this.

File: vulntotal/datasources/vulnerablecode.py [hunk 3]
```diff
@@ -94,13 +121,13 @@ class VCIOTokenError(Exception):
...
-
-              VCIO_TOKEN="OJ78Os2IPfM80hqVT2ek+1QnrTKvsX1HdOMABq3pmQd"
+    For example:
+        VCIO_TOKEN='OJ78Os2IPfM80hqVT2ek+1QnrTKvsX1HdOMABq3pmQd'
     """
+
     load_dotenv()
     vcio_token = os.environ.get("VCIO_TOKEN", None)
-    if not vcio_token:
+    if vcio_token is None:
```
Comment (keshav-space) [inline vulntotal/datasources/vulnerablecode.py:130]: This will not handle the case when vcio_token is empty string.

```python
>>> vcio_token = ""
>>> vcio_token is None
False
```

```suggestion
    if not vcio_token:
```
